doc: update CONTRIBUTING.md to address editing PRs

gibfahn opened the pull request:
##### Checklist
<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->

- [x] documentation is changed or added
- [x] commit message follows commit guidelines

##### Affected core subsystem(s)
<!-- Provide affected core subsystem(s) (like doc, cluster, crypto, etc). -->
doc

##### Description of change
<!-- Provide a description of the change below this comment. -->

Add more info about the contribution process after PR submission.

 - One of the things I frequently see with new contributors is that people don't appreciate that raising a PR is only half the battle, the other half is fixing nits. 
 - It's also a little discouraging to raise a PR and have changes suggested, especially if you weren't expecting it.
 - The other thing that often happens is people getting confused about how Github PRs work, I found it a bit unintuitive at first that the PR just points to your branch so you can just update your branch to update the PR. Obviously we don't want to start including Git tutorials, but this comes up often enough that I think it might be worth spelling it out.

Open to suggestions on any of this!

Trott commented on CONTRIBUTING.md:
```diff
@@ -194,10 +201,57 @@ $ git push origin my-branch
+
+### Step 7: Discuss and update
+
+You will probably get feedback or requests for changes to your Pull Request,
```
Nit: period/full stop instead of comma after `Pull Request` (and, therefore, `this` -> `This` in the next line).

Trott commented on CONTRIBUTING.md:
````diff
@@ -194,10 +201,57 @@ $ git push origin my-branch
+```
+
+**Important:** The `git push --force` command is one of the few ways to delete
+history in git, before you use it, make sure you understand the risks. If in
````
Nit: Period/full stop after `git` and therefore `before` -> `Before` also.

Trott commented on CONTRIBUTING.md:
```diff
@@ -194,10 +201,57 @@ $ git push origin my-branch
+
+**Important:** The `git push --force` command is one of the few ways to delete
+history in git, before you use it, make sure you understand the risks. If in
+doubt, you can always ask for guidance in the Pull Request or on IRC.
```
Nit: Link the word IRC to #node-dev IRC channel here the same way it is done in the main README?

gibfahn replied:
I think I used the right link, let me know.

Trott commented on CONTRIBUTING.md:
```diff
@@ -194,10 +201,57 @@ $ git push origin my-branch
+history in git, before you use it, make sure you understand the risks. If in
+doubt, you can always ask for guidance in the Pull Request or on IRC.
+
+If your Pull Request has gone quiet, post a comment in the pull request to ping
```
Nit: I don't much care whether we go with `Pull Request` or `pull request`, but please use one or the other exclusively. Both are used in this sentence. :-D 

Nit: `has gone quiet` may be a little colloquial. Maybe `If your Pull Request needs attention, post a comment...`?

Trott commented on CONTRIBUTING.md:
```diff
@@ -194,10 +201,57 @@ $ git push origin my-branch
+### Step 8: Landing
+
+Once your PR has been reviewed and approved by at least two Node.js
+Collaborators, and as long as there is consensus (no objections from a
```
Only one Collaborator need to approve it. (More is better, of course. And two CTC members are required for `semver-major` changes.)

Trott commented on CONTRIBUTING.md:
```diff
@@ -194,10 +201,57 @@ $ git push origin my-branch
+
+Once your PR has been reviewed and approved by at least two Node.js
+Collaborators, and as long as there is consensus (no objections from a
+collaborator or unresolved issues) a collaborator should merge the PR in for
```
Nit: I'd remove `or unresolved issues`

Nit: comma after the closing parenthesis

Nit: `should` -> `can` or `may` or even `will` but probably not `should`

Nit: Unless we explicitly indicate somewhere that `PR` stands for `Pull Request` (and we may--I didn't look at the whole doc), stick with `Pull Request` instead of `PR` here and below.

Nit: Leave off `in for you`

Trott commented on CONTRIBUTING.md:
```diff
@@ -194,10 +201,57 @@ $ git push origin my-branch
+Once your PR has been reviewed and approved by at least two Node.js
+Collaborators, and as long as there is consensus (no objections from a
+collaborator or unresolved issues) a collaborator should merge the PR in for
+you. Github often shows the PR as `Closed` at this point, but don't worry, if
```
Nit: Comma to period/full stop and change `if` to `If`.

gibfahn commented on CONTRIBUTING.md:
```diff
@@ -194,10 +201,57 @@ $ git push origin my-branch
+
+When someone suggests changes, you can address them by applying them to your
+branch. When your branch changes, Github will automatically update the PR.
+
```
Slightly amended to specify pushing their branch to their fork.

gibfahn commented:
@Trott I _think_ I addressed everything. Not entirely sure (e.g. about the IRC link). Let me know if you have any other suggestions, happy to keep revising! I think it's worth bikeshedding a fair bit for docs.

Is it worth adding that Pull Requests usually are left to stew for at least 48 hours? I was going to, but I feel like there's already a tonne of information there, and I don't want to overload people too much. 

In fact I was considering taking out the bit about needing two CTC LGTMs to keep it simpler.

Also maybe I should mention/explain the word LGTM in there?

Trott commented on CONTRIBUTING.md:
```diff
@@ -194,10 +201,60 @@ $ git push origin my-branch
+**Important:** The `git push --force-with-lease` command is one of the few ways
+to delete history in git. Before you use it, make sure you understand the risks.
+If in doubt, you can always ask for guidance in the Pull Request or on
+[IRC in the #node.js channel](https://webchat.freenode.net?channels=node.js&uio=d4).
```
I was thinking the #node-dev channel rather than the #Node.js channel.

Trott commented:
LGTM as is. I'll change my review status to Approved. Optional additional bike-shed points:

* I meant the #node-dev IRC channel and not the #Node.js channel, but either one is probably fine.

* On mentioning the 48 hours, I have no opinion. On the one hand, hey, it's another potentially-useful piece of information. On the other hand, too much information can make it harder for people to find the one or two pieces of information they _really_ need. So...  ¯\\_(ツ)_/¯

* On removing the bit about 2 CTC approvals required for semver-major: Yes, I agree, we don't _need_ that in this doc. 

* Defining LGTM: Probably a good idea.

gibfahn commented:
@Trott 
 - IRC channel: changed to node-dev
 - 48 hours: didn't seem necessary (on second thoughts)
 - CTC approvals: also unnecessary, removed
 - LGTM - defined (was going to add a bit saying you can use the Github approval button, but again, seems like TMI)

Still LGTY?

Trott commented:
Yes, still looks good to me.
